src/services/LogFileManager/index.ts

```diff
@@ -286,11 +289,12 @@ class LogFileManager {
      * Creates a RegExp object based on the given query string and options, and starts querying the
      * first log chunk.
      *
-     * @param queryString
-     * @param isRegex
-     * @param isCaseSensitive
+     * @param queryArgs
+     * @param queryArgs.queryString
+     * @param queryArgs.isRegex
+     * @param queryArgs.isCaseSensitive
      */
-    startQuery (queryString: string, isRegex: boolean, isCaseSensitive: boolean): void {
+    startQuery ({queryString, isRegex, isCaseSensitive}: QueryArgs): void {
```

Comment:
@junhaoliao I agree with the rabbit. RegExp will return syntax error if `regexPattern` is not a valid regex string.

Reply:
I see why we want to handle any invalid RegExp as the user could have manual errors in their input or they're inputting an incomplete query as they type. However, I don't agree with the Rabbit's code suggestion. When a RegExp error is identified, we should have:
1. Update the `queryId` in the service worker so that any scheduled search task would know a new query has started and wouldn't necessarily continue the query.
2. Signal the renderer so that the renderer understand a new query has started as well and clears any existing results.
which are what we have been doing.

How do you feel about adding a try-catch block around the RegExp construction which wraps the error with a "malformed query" message? 
<img width="1494" alt="image" src="https://github.com/user-attachments/assets/44ca322c-2302-4ce4-99cf-d2797954f603">